jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -159,47 +203,153 @@ def _on_spectral_unit_changed(self, *args):
                                self.spectral_unit.selected,
                                sender=self))
 
-    @observe('flux_unit_selected')
-    def _on_flux_unit_changed(self, *args):
+    @observe('flux_or_sb_selected', 'flux_unit_selected', 'sb_unit_selected')
+    def _on_flux_unit_changed(self, msg):
+        flux_or_sb = None
+        current_y = self.spectrum_viewer.state.y_display_unit
 
-        yunit = _valid_glue_display_unit(self.flux_or_sb_unit.selected, self.spectrum_viewer, 'y')
+        data_collection_unit = ''
+        # need to determine the input spectrum units to disable the additional
+        # drop down and possiblity of translations in Specviz.
+        if (
+            len(self.app.data_collection) > 0 and
+            self.app.data_collection[0] and
+            self.app.config == 'specviz'
+        ):
+            if check_if_unit_is_per_solid_angle(self.app.data_collection[0].get_object()._unit):
```

Comment:
If there is a safer way to attain the units from the data item that keeps us more open to enable free-form in the future .

Reply:
Accessing a hidden `_unit` (leading underscore usually hints that it is private and should not be used externally). I think you are basically trying to find flux unit, so any reason to not use a more public `get_object().flux.unit` (since this is limited to specviz so you are getting Spectrum1D back). Where did `_unit` come from anyway?

Reply:
Looks like it is buried within specutils class mixin. @rosteen , what is the difference between `Spectrum1D._unit` and `Spectrum1D.flux.unit`?

Reply:
It's not from the specutils class mixin as far as I can tell, I suspect it's inherited from a couple layers upstream. I'd just use `flux.unit`.